### PR TITLE
[13.0][ADD] vehicle

### PR DESCRIPTION
**Pull request by ChristianSantamaria**

### CODE REVIEW

File: vehicle/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+{
+    'name': 'Vehicle',
+    'author': 'Solvos',
+    'category': 'Fleet',
```
Comment (dalonsod) [inline vehicle/__manifest__.py]: Maybe better this known category: https://github.com/odoo/odoo/blob/13.0/addons/fleet/__manifest__.py#L7

File: vehicle/models/vehicle.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import fields, models, _, api
+
+
+class Vehicle(models.Model):
+    _name = "vehicle.vehicle"
+    _description = "Vehicle"
+
+    name = fields.Char(default="/", readonly=True)
```
Comment (dalonsod) [inline vehicle/models/vehicle.py]: Better this:
```suggestion
    name = fields.Char(default=_("New car"), readonly=True)
```

File: vehicle/models/vehicle.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Responsible user",
+        default=lambda self: self.env.user,
+        required=True)
+
+    @api.constrains('license_plate')
+    def _constrains_name(self):
+        # Example 
+        self.name = "VVV-" + self.license_plate
```
Comment (dalonsod) [inline vehicle/models/vehicle.py]: In this first version, it should be enough IMO:
```suggestion
        # Example 
        self.name = self.license_plate
```
We should improve this later

File: vehicle/security/ir.model.access.csv [hunk 4]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_vehicle_user,access.vehicle.vehicle.user,model_vehicle_vehicle,group_vehicle_user,1,0,1,0
```
Comment (dalonsod) [inline vehicle/security/ir.model.access.csv]: Write permission is needed by default. If you test with a User role, an error is fired when creating the record (probably constrains method write operation). In general, the user that creates a vehicle, should change it
```suggestion
access_vehicle_user,access.vehicle.vehicle.user,model_vehicle_vehicle,group_vehicle_user,1,1,1,0
```

File: vehicle/security/vehicle.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            <field name="model_id" ref="model_vehicle_vehicle" />
+            <field name="global" eval="True" />
+            <field name="domain_force">[('user_id','=',user.id)]</field>
+            <field name="perm_read" eval="0" />
+            <field name="perm_create" eval="0" />
+            <field name="perm_write" eval="1" />
+            <field name="perm_unlink" eval="0" />
+            <field name="groups" eval="[(4, ref('vehicle.group_vehicle_user'))]" />
+        </record>
+    </data>
```
Comment (dalonsod) [inline vehicle/security/vehicle.xml:40]: Record rule that invalidates the restricted one is missing (the typical `[(1, '=', 1)]`]). If you test with more than one user, you'll realize that nobody can change the others vehicles

File: vehicle/views/vehicle_views.xml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                </header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button
+                            name="toggle_active"
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-check"
+                        >
+                            <field name="active"/>
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml]: This button looks strange:

imagen

File: vehicle/views/vehicle_views.xml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        <field name="name">vehicle.vehicle.tree</field>
+        <field name="model">vehicle.vehicle</field>
+        <field name="arch" type="xml">
+            <tree string="Vehicles">
+                <field name="license_plate"/>
+                <field name="name"/>
+                <field name="user_id"/>
+            </tree>
+        </field>
+    </record>
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml:78]: I miss kanban view

File: vehicle/views/vehicle_views.xml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        <field name="model">vehicle.vehicle</field>
+        <field name="arch" type="xml">
+            <search string="Vehicle Search">
+                <filter
+                    string="Archived"
+                    name="inactive"
+                    domain="[('active','=',False)]"
+                />
+                <separator />
+                <field name="name" filter_domain="[('name', 'ilike', self)]" />
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml:26]: I miss a "My vehicles" filter

File: vehicle/views/vehicle_views.xml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                        </button>
+                    </div>
+                    <div class="oe_title">
+                        <h1 class="oe_title">
+                            <field name="name" />
+                        </h1>
+                    </div>
+                    <group>
+                        <field name="license_plate"/>
+                        <field name="user_id"/>
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml]: This field should be readonly for User roles, since selecting a different user should fail

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments. Great icon.

File: vehicle/models/vehicle.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import fields, models, _, api
+
+
+class Vehicle(models.Model):
+    _name = "vehicle.vehicle"
+    _description = "Vehicle"
+
+    name = fields.Char(default="/", readonly=True)
+    license_plate = fields.Char(required=True)
```
Comment (dalonsod) [inline vehicle/models/vehicle.py]: This field can be tracked, and should be unique

File: vehicle/views/vehicle_views.xml (hunk 6 above)
Comment (ChristianSantamaria) [reply vehicle/views/vehicle_views.xml]: I was looking for the bug in this field but apparently in the helpdesk_ticket addon in the stages form this visualization is also used and the same thing happens to it. We can add the field with a basic visualization in the form, what do you think?

Comment (dalonsod) [reply vehicle/views/vehicle_views.xml]: Please take a look at https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0 :

imagen

File: vehicle/views/vehicle_views.xml [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+<odoo>
+
+    <record id="view_vehicle_vehicle_search" model="ir.ui.view">
+        <field name="name">vehicle.vehicle.search</field>
+        <field name="model">vehicle.vehicle</field>
+        <field name="arch" type="xml">
+            <search string="Vehicle Search">
+                <filter
+                    string="My Vehicles"
+                    name="inactive"
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml]: Please fix this:
```suggestion
                    name="my_vehicles"
```

**Comment by dalonsod** [comment]
@ChristianSantamaria see changes made. If you have any question, please ask me.

Pending things:

- I've added tracking attributes, so the message wall is now needed.
- Could you add an empty button container, preventing future related buttons for vehicles? You can look at examples across Odoo.
- There's a peding problem in kanban view: clicking in a kanban does not take effect (accessing to the vehicle).

**Comment by dalonsod** [comment]
And also,

- Duplicate function is not working, because `license_plate` field is not copied (good) but it's required. I think the best solution here is disabling copy functionality, like e.g. https://github.com/odoo/odoo/blob/13.0/addons/hr_attendance/models/hr_attendance.py#L103

**Comment by ChristianSantamaria** [comment]
The changes have been added and I have created a new icon:
imagen

I tried your idea of adding a license plate but I was not convinced, so I put a truck. What do you think?

File: vehicle/views/vehicle_views.xml [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+                <attribute name="readonly">1</attribute>                   
+            </field>                
+        </field>
+    </record>
+    <record id="view_vehicle_vehicle_form_manager" model="ir.ui.view">
+        <field name="name">vehicle.vehicle.form (user_id manager)</field>
+        <field name="model">vehicle.vehicle</field>
+        <field name="inherit_id" ref="vehicle.view_vehicle_vehicle_form" />
+        <field name="groups_id" eval="[(4, ref('vehicle.group_vehicle_manager'))]" />
+        <field name="priority" eval="25"/>
```
Comment (dalonsod) [inline vehicle/views/vehicle_views.xml:127]: Added this lower priority for this view. With this addtion, Manager role (which is User too) will always have editable permissions. With the same priority as User view, there was a random behavior (if user view was applied after, an issue was raised)

**Review by dalonsod — APPROVED** [review]
The addon is ready! @ChristianSantamaria  please add i18n and html readme files, and squash commits

**Pull request merged**